private/pkg/storage/storagegit/provider.go

```diff
@@ -0,0 +1,41 @@
+// Copyright 2020-2023 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storagegit
+
+import (
+	"github.com/bufbuild/buf/private/pkg/git"
+	"github.com/bufbuild/buf/private/pkg/storage"
+)
+
+type provider struct {
```

Comment:
You'll likely have a `symlinks` option and `symlinksIfSupported` option added to `NewReadBucket`, similar to OS...but there might be a reason that symlinks are safe with git, but I'm not sure without research

Reply:
Yeah I looked into it, and we have the same issues, Git will restore symlinks to the host fs and they may link to files outside of the repo, so same security issues there. I erroneously thought we always enable them, didn't notice the `--disable-symlinks` option. I'll add that back in.

Reply:
Done